pkg/scanner/sbom/scan.go

```diff
@@ -0,0 +1,242 @@
+package sbom
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/google/wire"
+	"golang.org/x/exp/slices"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/applier"
+	ftypes "github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/detector/library"
+	ospkgDetector "github.com/aquasecurity/trivy/pkg/detector/ospkg"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+var (
+	pkgTargets = map[string]string{
+		ftypes.PythonPkg: "Python",
+		ftypes.GemSpec:   "Ruby",
+		ftypes.NodePkg:   "Node.js",
+		ftypes.Jar:       "Java",
+	}
+)
+
+// SuperSet binds dependencies for Local scan
+var SuperSet = wire.NewSet(
+	applier.NewApplier,
+	wire.Bind(new(Applier), new(applier.Applier)),
+	ospkgDetector.SuperSet,
+	wire.Bind(new(OspkgDetector), new(ospkgDetector.Detector)),
+	NewScanner,
+)
```

Comment:
Local/scanner processes are diverted due to lack of understanding around the wire.